homeassistant/components/squeezebox/button.py

```diff
@@ -0,0 +1,199 @@
+"""Platform for button integration for squeezebox."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+import logging
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import (
+    CONNECTION_NETWORK_MAC,
+    DeviceInfo,
+    format_mac,
+)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import SqueezeboxConfigEntry
+from .const import DOMAIN, SIGNAL_PLAYER_DISCOVERED
+from .coordinator import SqueezeBoxPlayerUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+HARDWARE_MODELS_WITH_SCREEN = [
+    "Squeezebox Boom",
+    "Squeezebox Radio",
+    "Transporter",
+    "Squeezebox Touch",
+    "Squeezebox",
+    "SliMP3",
+    "Squeezebox 1",
+    "Squeezebox 2",
+    "Squeezebox 3",
+]
+
+
+@dataclass(frozen=True, kw_only=True)
+class SqueezeboxButtonEntityDescription(ButtonEntityDescription):
+    """Squeezebox Button description."""
+
+    press_action: str
+    models: list[str] = field(default_factory=list)
+
+
+BUTTON_ENTITIES: tuple[SqueezeboxButtonEntityDescription, ...] = (
+    SqueezeboxButtonEntityDescription(
+        key="preset_1",
+        name="Preset 1",
+        translation_key="preset_1",
+        press_action="preset_1.single",
+    ),
+    SqueezeboxButtonEntityDescription(
+        key="preset_2",
+        name="Preset 2",
+        translation_key="preset_2",
+        press_action="preset_2.single",
+    ),
```

Comment:
```suggestion
    SqueezeboxButtonEntityDescription(
        key="preset_1",
        name="Preset 1",
        translation_key="preset_1",
        press_action="preset_1.single",
    ),
    SqueezeboxButtonEntityDescription(
        key="preset_2",
        name="Preset 2",
        translation_key="preset_2",
        press_action="preset_2.single",
    ),
```
I think we can just create one for every number in an inline forloop

Reply:
I can't figure out how to do that when I'm creating class instances - I thought I could do
```python
    SqueezeboxButtonEntityDescription(
        key=f"preset_{val}",
        translation_key=f"preset_{val}",
        press_action=f"preset_{val}.single",
    ) for val in range(1,6)
```
but that doesn't work.  Any ideas?